Readme: updated (#209)
numtide/treefmt: README.md

**Status: beta**

`treefmt` applies all the needed formatters to your project with one command line.

## Motivation

Before making contributions to any project, it’s common to get your code formatted according to the project’s standards. This task seems trivial from the first sight — you can simply set up the required language formatter in your IDE. But contributing to multiple projects requires more effort: you need to change the code formatter configs each time you switch between projects, or call formatters manually.

Formatting requires less effort if a universal formatter for multiple languages is in place, which is also project-specific.

## About treefmt

> `treefmt` runs all your formatters with one command. It’s is easy to configure and fast to execute.

Reviewer: ```suggestion
`treefmt` runs all your formatters with one command. It’s easy to configure and fast to execute.
```

[![asciicast](https://asciinema.org/a/cwtaWUTdBa8qCKJVp40bTwxf0.svg)](https://asciinema.org/a/cwtaWUTdBa8qCKJVp40bTwxf0)

Its main features are:

- Providing a unified CLI and output: You don’t need to remember which formatters are required for each project. Once you specify the formatters in the config file, you can trigger all of them with one command and get a standardized output.
- Running all the formatters in parallel: A standard script loops over your folders and runs each formatter consequentially. In contrast, `treefmt` runs formatters in parallel. This way, the formatting job takes less time.
- Caching the changed files: When formatters are run in a script, they process all the files they encounter, no matter whether the code has changed. This unnecessary work can be eliminated if only the changed files are formatted. `treefmt` caches the changed files and marks them for re-formatting.

> Just type `treefmt` in any folder to reformat the whole code tree. All in all, you get a fast and simple formatting solution. Find the project wiki [here](https://github.com/numtide/treefmt/wiki).

Reviewer: ```suggestion
Just type `treefmt` in any folder to reformat the whole code tree. All in all, you get a fast and simple formatting solution.
```
The transition in the sentence is a bit weird. The main use of the wiki is to share formatter formulas, so it could be introduced somewhere down the line?

## Installation

You can install the tool by downloading the binary. Find the binaries for different architectures [here](https://github.com/numtide/treefmt/releases). Otherwise, you can install the package from the source code — either with [cargo](https://github.com/rust-lang/cargo), or with help of [nix](https://github.com/NixOS/nix). We describe the installation process in detail in the [wiki](https://github.com/numtide/treefmt/wiki).

## Usage

In order to use `treefmt` in your project, make sure the config file `treefmt.toml` is present in the root folder and is edited to your needs. You can generate it with:

```
$ treefmt --init
```

You can run `treefmt` in your project root folder like this:

```
$ treefmt
```

To explore the tool’s flags and options, type:

> ```

Reviewer: ```suggestion
```console
```
that helps the highlighter

$ treefmt --help
```

> You can also integrate `treefmt` into your CI.

Reviewer: ```suggestion
```
For now let's remove it and introduce a chapter in the docs specifically about CI support.

## Configuration

Fomatters are specified in the config file `treefmt.toml`, which is usually located in the project root folder. The generic way to specify a formatter is like this:

```
[formatter.<name>]
command = "<formatter-command>"
options = [“<formatter-option-1>”...]
includes = ["<glob>"]
```

For example, if you want to use [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) on your Nix project and rustfmt on your Rust project, then `treefmt.toml` will look as follows:

```
[formatter.nix]
command = "nixpkgs-fmt"
includes = ["*.nix"]

[formatter.rust]
command = "rustfmt"
options = ["--edition", "2018"]
includes = ["*.rs"]
```

> Before specifying the formatter in the config, make sure it’s installed. See the [wiki](https://github.com/numtide/treefmt/wiki) for more examples.

Reviewer: ```suggestion
Before specifying the formatter in the config, make sure it’s installed.

To find and share existing formatter recipes, take a look at the [wiki](https://github.com/numtide/treefmt/wiki).

If you are a Nix user, you might also be interested in [treefmt-nix](https://github.com/numtide/treefmt-nix) to use Nix to configure and bring in formatters.
```

## Compatibility

`Treefmt` works with any formatter that adheres to the [following specification](https://github.com/renoire/treefmt/blob/master/docs/formatters-spec.md). For instance, you can go for:

- [clang-format](https://clang.llvm.org/docs/ClangFormat.html) for Java
- gofmt for Golang
- Prettier for JavaScript/HTML/CSS

Find the full list of supported formatters [here](https://numtide.github.io/treefmt/formatters.html).

> ## Upcoming features

Reviewer: ```suggestion
## Upcoming features

This project is still pretty new. Down the line we also want to add support for:
```
